lib/lint.ml

```diff
@@ -102,11 +102,24 @@ let lint_opam_doc pkg =
   0
 
 let lint_opam_home_and_dev pkg =
-  lint_res
-    ~msgf:(fun l ->
-      l "opam fields %a and %a can be parsed by dune-release" pp_field
-        "homepage" pp_field "dev-repo")
-    (Pkg.distrib_user_and_repo pkg)
+  ( match Pkg.distrib_uri pkg with
+  | Error _ ->
+      report_status `Fail (fun l ->
+          l
+            "Fields homepage and dev-repo are not valid github URIs, \
+             dune-release will not be able to automatically publish the \
+             package.")
+  | Ok "" ->
+      report_status `Fail (fun l ->
+          l "Skipping homepage and dev-repo fields linting, no field found")
+  | Ok _ ->
+      let pass = R.is_ok (Pkg.Github.distrib_uri pkg) in
```

Comment:
Is this here in case the `--dist-uri` option was passed or is there another reason?

Reply:
This just indicates whether we might face an issue later on if the distrib uri is not a github uri, but without failing.